### PR TITLE
Initial Pull Request for BitShares on Ubuntu

### PR DESCRIPTION
**Pull request by ryanRfox**
I hope this all goes to plan.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@ryanRfox__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

File: bitshares-ubuntu-vm/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+      "metadata": {
+        "description": "Size of VM"
+      }
+    }
+  },
+  "variables": {
+    "imagePublisher": "Canonical",
+    "imageOffer": "UbuntuServer",
+    "OSDiskName": "BitSharesOSDisk",
+    "imageSKU": "15.10",
```
Comment (singhkay) [inline bitshares-ubuntu-vm/azuredeploy.json]: Can this run on 14.04 LTS? 15.10 will EOL'd in a few months and then the image will be removed from Azure causing this template to break

File: bitshares-ubuntu-vm/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    "nicName": "BitSharesNic",
+    "addressPrefix": "10.0.0.0/16",
+    "subnetName": "BitSharesSubnet",
+    "subnetPrefix": "10.0.0.0/24",
+    "storageAccountType": "Standard_LRS",
+    "storageAccountName": "[replace(replace(tolower(concat('bitsharessa', uniquestring(resourceGroup().id))), '-',''),'.','')]",
```
Comment (singhkay) [inline bitshares-ubuntu-vm/azuredeploy.json]: Use the unique string first, it helps with spreading storage accounts from this template on different clusters. Also you don't need to replace and toLower output of uniquestring. It already is compatible with storage account names.

File: bitshares-ubuntu-vm/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+            "enabled": "true",
+            "storageUri": "[concat('http://',variables('storageAccountName'),'.blob.core.windows.net')]"
+          }
+        }
+      }
+    },  
+    {
+      "type": "Microsoft.Compute/virtualMachines/extensions",
+      "name": "[concat(variables('vmName'),'/newuserscript')]",
+      "apiVersion": "2015-05-01-preview",
```
Comment (singhkay) [inline bitshares-ubuntu-vm/azuredeploy.json]: apiVersion variable not used here

File: bitshares-ubuntu-vm/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+      "location": "[resourceGroup().location]",
+      "dependsOn": [
+        "[concat('Microsoft.Compute/virtualMachines/', variables('vmName'))]"
+      ],
+      "properties": {
+        "publisher": "Microsoft.OSTCExtensions",
+        "type": "CustomScriptForLinux",
+        "typeHandlerVersion": "1.2",
```
Comment (singhkay) [inline bitshares-ubuntu-vm/azuredeploy.json]: Use 1.4 here, also add the autoUpgradeMinorVersion property

File: bitshares-ubuntu-vm/azuredeploy.json (hunk 1 above)
Comment (ryanRfox) [inline bitshares-ubuntu-vm/azuredeploy.json]: This template will not currently build on 14 LTS due to Boost version incompatibility (need 1.58+). I plan to continue to maintain the template and use then current builds (15+).

Will this be acceptable?

Cheers,
Fox

On Mar 5, 2016, at 8:50 AM, Kay Singh <notifications@github.com<mailto:notifications@github.com>> wrote:


In bitshares-ubuntu-vm/azuredeploy.json<https://github.com/Azure/azure-quickstart-templates/pull/1335#discussion_r55094039>:

> +        "Standard_A2",
> +        "Standard_A3",
> +        "Standard_D1_V2",
> +        "Standard_D2_V2",
> +        "Standard_D3_V2"
> +      ],
> +      "metadata": {
> +        "description": "Size of VM"
> +      }
> +    }
> +  },
> +  "variables": {
> +    "imagePublisher": "Canonical",
> +    "imageOffer": "UbuntuServer",
> +    "OSDiskName": "BitSharesOSDisk",
> +    "imageSKU": "15.10",


Can this run on 14.04 LTS? 15.10 will EOL'd in a few months and then the image will be removed from Azure causing this template to break

-

Comment (singhkay) [inline bitshares-ubuntu-vm/azuredeploy.json]: That sounds good :+1:

**Pull request closed without merging**